Formula/vtk.rb

```diff
@@ -47,20 +47,24 @@ class Vtk < Formula
   uses_from_macos "zlib"
 
   on_macos do
-    depends_on "llvm" => :build if DevelopmentTools.clang_build_version == 1316 && Hardware::CPU.arm?
+    on_arm do
+      depends_on "llvm" => :build if DevelopmentTools.clang_build_version == 1316
+    end
   end
 
   on_linux do
     depends_on "libaec"
     depends_on "mesa-glu"
   end
 
-  fails_with gcc: "5"
-
   # clang: error: unable to execute command: Segmentation fault: 11
   # clang: error: clang frontend command failed due to signal (use -v to see invocation)
   # Apple clang version 13.1.6 (clang-1316.0.21.2)
-  fails_with :clang if DevelopmentTools.clang_build_version == 1316 && Hardware::CPU.arm?
+  on_arm do
+    fails_with :clang if DevelopmentTools.clang_build_version == 1316
```

Comment:
Could this not be also in the above `on_macos` + `on_arm` block? I think it is macOS-specific.